corebehrt/common/utils.py

```diff
@@ -100,80 +53,109 @@ def load_tensor(filename, required=False):
-        outcomes = load_tensor(f'{mode}_outcomes.pt')
-        censor_outcomes = load_tensor(f'{mode}_censor_outcomes.pt')
-        vocabulary = load_tensor('vocabulary.pt')
+
+        features = load_tensor(f"{mode}_features.pt", required=True)
+        pids = load_tensor(f"{mode}_pids.pt", required=True)
+        outcomes = load_tensor(f"{mode}_outcomes.pt")
+        censor_outcomes = load_tensor(f"{mode}_censor_outcomes.pt")
+        vocabulary = load_tensor("vocabulary.pt")
         return cls(features, pids, outcomes, censor_outcomes, vocabulary, mode=mode)
-    
+
     def check_lengths(self):
         """Check that all features have the same length"""
         for key, values in self.features.items():
-            assert len(values) == len(self.pids), f"Length of {key} does not match length of pids"
+            assert len(values) == len(
+                self.pids
+            ), f"Length of {key} does not match length of pids"
         if self.outcomes is not None:
-            assert len(self.outcomes) == len(self.pids), "Length of outcomes does not match length of pids"
+            assert len(self.outcomes) == len(
+                self.pids
+            ), "Length of outcomes does not match length of pids"
         if self.censor_outcomes is not None:
-            assert len(self.censor_outcomes) == len(self.pids), "Length of censor outcomes does not match length of pids"
+            assert len(self.censor_outcomes) == len(
+                self.pids
+            ), "Length of censor outcomes does not match length of pids"
 
-    def split(self, val_split: float)->Tuple['Data', 'Data']:
+    def split(self, val_split: float) -> Tuple["Data", "Data"]:
         """Split data into train and validation. Returns two Data objects"""
         train_indices, val_indices = self._get_train_val_splits(val_split)
 
-        train_data = self.select_data_subset_by_indices(train_indices, 'train')
-        val_data = self.select_data_subset_by_indices(val_indices, 'val')
+        train_data = self.select_data_subset_by_indices(train_indices, "train")
+        val_data = self.select_data_subset_by_indices(val_indices, "val")
         return train_data, val_data
-    
-    def select_data_subset_by_indices(self, indices: list, mode:str ='')->'Data':
-        return Data(features={key: [values[i] for i in indices] for key, values in self.features.items()}, 
-                        pids=[self.pids[i] for i in indices],
-                        outcomes=[self.outcomes[i] for i in indices] if self.outcomes is not None else None,
-                        censor_outcomes=[self.censor_outcomes[i] for i in indices] if self.censor_outcomes is not None else None,
-                        vocabulary=self.vocabulary,
-                        mode=mode)
-    
-    def select_data_subset_by_pids(self, pids: list, mode:str='')->'Data':
+
+    def select_data_subset_by_indices(self, indices: list, mode: str = "") -> "Data":
```

Comment:
This should maybe be a part of the `Data` class as a `subset` method, as it only functions on this

Reply:
We won't need the method in the future